rfcs/rfc-005-branches-tags-and-channels.md

```diff
@@ -0,0 +1,52 @@
+```
+========================= master branch (edge channel) =======================>
+         \                      \                     \
+          \___v0.7.0 tag         \                     \
+           \                      \         v0.9.0 tag__\
+            \          v0.8.0 tag__\                     \
+ v0.7.1 tag__\                      \                 v0.9 branch (beta channel)
+              \___v0.7.2 tag         \___v0.8.1 tag
+               \                      \
+                \                      \
+           v0.7 branch         v0.8 branch (stable channel)
+
+```
+
+## Branches and Tags
+
+### master branch
+All new development occurs on the `master` branch.
+
+Bug fixes that affect a `vX.Y` branch are first made on `master`.  Once the bug
```

Comment:
I'm a bit concerned with using the same naming convention for branch`vX.Y` and tag `vX.Y.Z`. What happens if we want to create a `vX.Y.Z` branch from the `vX.Y.Z` tag? Maybe it's technically possible, but will that create unnecessary confusion? For example, if we want to discuss `v8.0` or `v8.0.0` how exactly should we know if we're talking about a branch or a tag? Count the digits? Always write the word "branch" or "tag" just after? Instead, is there a popular naming convention for release branches that we can adopt? Maybe a `rel-` or `release-` prefix?

Reply:
I would hope we never have to create a branch from a `vX.Y.Z` tag.  Can you think of a realistic example?  

Talking about the X.Y milestone seems nice to me, saying X.Y.Z (0.7.0, 0.8.0) as we have in the past is unnecessarily verbose.  The first release on the X.Y milestone is just X.Y.0, then X.Y.1.  You just *know* that X.Y is a branch and X.Y.Z is a tag.  These are things we'll be saying every day so we don't want to encumber them with additional prefixes or suffixes IMO.

Reply:
I was thinking *every* branch would be created from tags, `.0` tags.  But even if not lazy branching, let's say v0.7.2 adds good things and bad things and we'd want to move 2 different parties forward with both a v0.7.1.1 and a v0.7.3.

What if that v0.7 branch was named v0.7.x?